Common/ExecutionModel/vtkThreadedImageAlgorithm.cxx

```diff
@@ -25,12 +25,105 @@
 #include "vtkPointData.h"
 #include "vtkStreamingDemandDrivenPipeline.h"
 
+#include "vtkExtentTranslator.h"
+
+
+
+//---- option 2
+
+     #include <tbb/parallel_for.h>
+#include <tbb/blocked_range.h>
+#include <tbb/blocked_range3d.h>
+#include <tbb/blocked_range2d.h>
+#include <tbb/task_scheduler_init.h>
+//-- option 1
+#include "vtkSMPTools.h"
+
+
+
+
+
+
+struct vtkImageThreadStruct
+{
+vtkThreadedImageAlgorithm *Filter;
+vtkInformation *Request;
+vtkInformationVector **InputsInfo;
+vtkInformationVector *OutputsInfo;
+vtkImageData   ***Inputs;
+vtkImageData   **Outputs;
+};
+
+class apply_transform{
```

Comment:
VTK Code Style (see https://docs.google.com/document/d/1nzinw-dR5JQRNi_gb8qwLL5PnkGMK2FETlQGLr10tZw/edit):
* class names use CamelCase rather than underscores (e.g. vtkImageThreadStruct)
* braces for class, method, and function definitions always go on the next line (traditional C/C++ style), not on the same line (which would be Java style)
* all member variables should be protected: or private:, and should be capitalized
* member variables go at the bottom of the class definition, because they are part of the class implementation rather than part of the class interface (interface methods are more important to people reading your code, hence they go first).